srcdeps-maven-plugin/src/main/java/org/srcdeps/mvn/plugin/SrcdepsInitMojo.java

```diff
@@ -0,0 +1,516 @@
+        private final Map<String, Map<String, Integer>> urlGroupIdMap = new HashMap<>();
+
+        private ScmRepositoryIndex(MavenSession session, RepositorySystemSession repoSession,
+                ArtifactFactory repositorySystem, ProjectBuilder projectBuilder, Log log,
+                Set<org.srcdeps.core.Scm> scms) {
+            super();
+            this.session = session;
+            this.repoSession = repoSession;
+            this.repositorySystem = repositorySystem;
+            this.projectBuilder = projectBuilder;
+            this.log = log;
+            this.remoteRepos = RepositoryUtils.toRepos(session.getProjectBuildingRequest().getRemoteRepositories());
+            this.scms = scms;
+        }
+
+        /**
+         * Associate the given {@code url} with the given {@code gav}
+         *
+         * @param url
+         *            a SCM URL cleaned from any {@code artifactId} suffixes
+         * @param gav
+         *            the {@link Gav} to associate with the given {@code url}
+         */
+        private void add(String url, Gav gav) {
+            final String groupId = gav.getGroupId();
+            Set<Gav> urlGavs = urlGavMap.get(url);
+            if (urlGavs == null) {
+                urlGavs = new HashSet<>();
+                urlGavMap.put(url, urlGavs);
+            }
+            urlGavs.add(gav);
+
+            Map<String, Integer> groupIdCntMap = urlGroupIdMap.get(url);
+            if (groupIdCntMap == null) {
+                groupIdCntMap = new HashMap<>();
+                urlGroupIdMap.put(url, groupIdCntMap);
+                Integer cnt = Integer.valueOf(1);
+                groupIdCntMap.put(groupId, cnt);
+            } else {
+                Integer cnt = groupIdCntMap.get(groupId);
+                cnt = cnt == null ? Integer.valueOf(0) : cnt;
+                groupIdCntMap.put(groupId, cnt.intValue() + 1);
+            }
+
+            Set<String> urls = groupIdUrlMap.get(groupId);
+            if (urls == null) {
+                urls = new HashSet<>();
+                groupIdUrlMap.put(groupId, urls);
+            }
+            urls.add(url);
+        }
+
+        /**
+         * Find the SCM URL for the given {@code g, a, v} triple and store the association for the later retrieval via
+         * {@link #createSortedScmRepositoryMap()}.
+         *
+         * @param g
+         *            {@code groupId}
+         * @param a
+         *            {@code artifactId}
+         * @param v
+         *            {@code version}
+         * @param failOnUnresolvable
+         *            see {@link SrcdepsInitMojo#failOnUnresolvable}
+         * @throws MojoExecutionException
+         */
+        public void addGav(String g, String a, String v, boolean failOnUnresolvable) throws MojoExecutionException {
+            final Gav gav = new Gav(g, a, v);
+            if (!seenGavs.contains(gav)) {
+                seenGavs.add(gav);
+                log.info("GAV: " + gav);
```

Comment:
The stock Maven offers zero means to filter debug output. One either has the full debug output of all running Maven + plugins code or no debug output at all. That leads me to start with info level logging everywhere, and when my code matures, I am gradually hiding the statements by moving them to debug or trace level. 

Actually, what I said in the prev. paragraph is not true anymore. Maven is slowly moving to SLF4j and one could use SLF4j loggers in the plugin code (in the very same way as I already do in all other non-plugin code here and srcdeps-core). I should migrate the code here in the plugin and I should learn how to control the per-topic levels. There is some info over here: https://maven.apache.org/maven-logging.html

Let's do it later.